### PR TITLE
Refactor model.rs & QueryPlanner related stuff

### PR DESCRIPTION
**Pull request by cecton**
## Changes

- ~~Cleanup snapshots: some snapshots files were unused and some had path not update. I made another PR for this #234~~
- Parse planner result as a `PlannerResult` instead of `QueryPlanner`: this new `PlannerResult` enum can detect only the kind `QueryPlanner` and will return a specific error on parsing if something else is returned by the router bridge.
- Moved most of the stuff in `model.rs` to `query_planner/mod.rs` and make them private
- Make `Selection` private and move all the logic around "selection" of the query planner to `selection.rs` (it's kinda big, it deserves its own file)
- Moved function `select()` on `Response` to `selection.rs` and make it private
- Deleted `apollo-router/tests/fixtures/query_plan.json` because it was not used anymore

Related to #234 
Related to #233

### CODE REVIEW

**Comment by Geal** [comment]
could we hold off on big file movements for a while after this PR lands? It has been happening often recently, and it is getting hard to follow, especially if we are trying to modify the same files in other pull requests.

It also does not help for reviews when there are other changes during the move. Like, in this commit https://github.com/apollographql/router/pull/233/commits/7524bc52f4ba55a29cdd1cb72b273ac4eb9018a4, ath the same time there's the change to use `QueryPlanResult` and deleting the model.rs file.

**Review by Geal — APPROVED** [review]

**Comment by cecton** [comment]
@Geal all things are interconnected. I made the list of changes I did in the PR description.

**Comment by cecton** [comment]
@Geal it's mostly just moving things around and replacing some `pub` by `pub(crate)`.

**Comment by Geal** [comment]
yes I am not saying we should never move things around. But in the past 2 weeks there's been #203 #205 #166 #153 and #140 moving code around. This is a lot, and I wish the code organisation would settle down a bit, because it's hard to review, it's annoying to integrate when we're working concurrently, and going through the code's history and understand past changes is arduous

**Comment by cecton** [comment]
Yes true...

Though #203 is a noop. When I updated my branches after that one got merged, the merge was almost automatic locally. Maybe you are rebasing a lot?

**Comment by Geal** [comment]
I'm a git monkey, I jump from branch to branch all day 🙈
But also: looking through git history is a very regular task.
And with this team's size it will often happen that multiple people are working on the same files at the same time. I would like you to be mindful of that

**Review by BrynCooke — APPROVED** [review]

**Comment by cecton** [comment]
Merging this for now as it can conflict easily with other PRs. Don't hesitate to give feedback later
